src/attribute.cpp

```diff
@@ -27,6 +43,175 @@ Integer::Integer(uint8_t type, const uint8_t* data, size_t size)
             + data[3];
 }
 
+Integer::Integer(uint8_t type, uint32_t value)
+    : Attribute(type),
+      m_value(value)
+{
+}
+
+std::vector<uint8_t> Integer::toVector(const std::string& secret, std::array<uint8_t, 16> auth) const
+{
+    std::vector<uint8_t> attribute(6);
+    attribute[0] = type();
+    attribute[1] = 6;
+    attribute[2] = m_value / (1 << 24);
+    attribute[3] = m_value / (1 << 16);
+    attribute[4] = m_value / (1 << 8);
+    attribute[5] = m_value;
+    return attribute;
+}
+
+IpAddress::IpAddress(uint8_t type, const uint8_t* data, size_t size)
+        : Attribute(type)
+{
+    if (size != 4)
+        throw std::runtime_error("Invalid integer attribute size. Should be 4, actual size is " + std::to_string(size));
+
+    for (size_t i = 0; i < size; ++i)
+        m_value[i] = data[i];
+}
+
+IpAddress::IpAddress(uint8_t type, std::array<uint8_t, 4> address)
+    : Attribute(type),
+      m_value(address)
+{
+}
+
+std::vector<uint8_t> IpAddress::toVector(const std::string& secret, std::array<uint8_t, 16> auth) const
+{
+    std::vector<uint8_t> attribute;
+    for (size_t i = 0; i < m_value.size(); ++i)
+        attribute.push_back(m_value[i]);
+
+    auto it = attribute.begin();
+    it = attribute.insert(it, attribute.size() + 2);
+    it = attribute.insert(it, type());
+
+    return attribute;
+}
+
+Encrypted::Encrypted(uint8_t type, const uint8_t* data, size_t size, std::string secret, std::array<uint8_t, 16> auth)
+        : Attribute(type)
+{
+    if (size > 128)
+        throw std::runtime_error("Invalid encrypted attribute size. Should be max 128 bytes, actual size is " + std::to_string(size));
+
+    std::vector<uint8_t> value(size);
+
+    size_t j = 16;
+    while (j <= size)
+    {
+        std::vector<uint8_t> buffer(16 + secret.length());
+
+        for (size_t i = 0; i < secret.length(); ++i)
+            buffer[i] = secret[i];
+
+        if (j == 16)
+        {
+            for (size_t i = 0; i < 16; ++i)
+                buffer[i + secret.length()] = auth[i];
+        }
+        else
+        {
+            for (size_t i = 0; i < 16; ++i)
+                buffer[i + secret.length()] = data[i + j - 32];
+        }
+        std::array<uint8_t, 16> md;
+
+        MD5(buffer.data(), buffer.size(), md.data());
+
+        for (size_t i = 0; i < 16; ++i)
+            value.push_back(data[i + j - 16] ^ md[i]);
+
+        j += 16;
+    }
+    m_value.assign(value.begin(), value.end());
+}
+
+Encrypted::Encrypted(uint8_t type, std::string password)
+    : Attribute(type),
+      m_value(password)
+{
+}
+
+std::vector<uint8_t> Encrypted::toVector(const std::string& secret, std::array<uint8_t, 16> auth) const
+{
+    if (m_value.length() > 128)
```

Comment:
We assume that internal representation (`m_value`) is always correct. Otherwise it would not be constructed. So this check is meaningles. But see my question above regarding the size constraint.